### PR TITLE
light: remove test panic

### PR DESCRIPTION
**Pull request by tychoish**
I left this in a pervious pass, because there's only one panic to
remove, but the size of the diff is obviously quite large.

### CODE REVIEW

File: light/client_test.go [hunk 1]
```diff
@@ -26,1092 +26,1084 @@ const (
...
-		// interim header (3/3 signed)
-		2: h2,
-		// last header (3/3 signed)
-		3: h3,
-	}
-	l1 = &types.LightBlock{SignedHeader: h1, ValidatorSet: vals}
-	l2 = &types.LightBlock{SignedHeader: h2, ValidatorSet: vals}
-	l3 = &types.LightBlock{SignedHeader: h3, ValidatorSet: vals}
-)
+var bTime, _ = time.Parse(time.RFC3339, "2006-01-02T15:04:05Z")
```
Comment (creachadair) [inline light/client_test.go:29]: Optional:
```suggestion
var bTime time.Time

func init() {
	var err error
	bTime, err = time.Parse(time.RFC3339, "2006-01-02T15:04:05Z")
	if err != nil {
		panic(err)
	}
}
```

**Review by creachadair — APPROVED** [review]

File: light/helpers_test.go [hunk 2]
```diff
@@ -74,7 +76,7 @@ func (pkz privKeys) ToValidators(init, inc int64) *types.ValidatorSet {
 }
 
 // signHeader properly signs the header with all keys from first to last exclusive.
-func (pkz privKeys) signHeader(header *types.Header, valSet *types.ValidatorSet, first, last int) *types.Commit {
+func (pkz privKeys) signHeader(t testing.TB, header *types.Header, valSet *types.ValidatorSet, first, last int) *types.Commit {
```
Comment (williambanfield) [inline light/helpers_test.go:79]: mark as helper?

File: light/helpers_test.go [hunk 3]
```diff
@@ -143,26 +143,26 @@ func genHeader(chainID string, height int64, bTime time.Time, txs types.Txs,
 }
 
 // GenSignedHeader calls genHeader and signHeader and combines them into a SignedHeader.
-func (pkz privKeys) GenSignedHeader(chainID string, height int64, bTime time.Time, txs types.Txs,
+func (pkz privKeys) GenSignedHeader(t testing.TB, chainID string, height int64, bTime time.Time, txs types.Txs,
```
Comment (williambanfield) [inline light/helpers_test.go:146]: mark as helper?

File: light/helpers_test.go [hunk 4]
```diff
@@ -143,26 +143,26 @@ func genHeader(chainID string, height int64, bTime time.Time, txs types.Txs,
...
 	return &types.SignedHeader{
 		Header: header,
-		Commit: pkz.signHeader(header, valset, first, last),
+		Commit: pkz.signHeader(t, header, valset, first, last),
 	}
 }
 
 // GenSignedHeaderLastBlockID calls genHeader and signHeader and combines them into a SignedHeader.
-func (pkz privKeys) GenSignedHeaderLastBlockID(chainID string, height int64, bTime time.Time, txs types.Txs,
+func (pkz privKeys) GenSignedHeaderLastBlockID(t testing.TB, chainID string, height int64, bTime time.Time, txs types.Txs,
```
Comment (williambanfield) [inline light/helpers_test.go:157]: mark as helper?

**Pull request merged**